### PR TITLE
Construct ObservationId directly from a value

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

This PR changes the constructor of `ObservationId` to take a `double value` directly, instead of a type that has a `value()` member function. This allows me to remove the wrapper around `size_t` that I used for the `LinearSolver::IterationId`. I haven't propagated the change to the other places that construct `ObservationId` yet, so opening this as a draft PR to confirm the design.

### Types of changes:

- [x] New feature

### Component:

- [x] Code

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by wthrowe** [comment]
Yes, this looks good.

**Comment by nilsdeppe** [comment]
I'll review, but might not have time until Tuesday or Wednesday :)

**Comment by nilsvu** [comment]
@wthrowe How should we handle time here? I can think of the following options:
- Call `time.value()` when constructing `ObservationId`s (I saw that you removed the `TimeValue` compute tag in some PR, which could have been used for this?)
- Make `TimeId` implicitly convertible to `double` so that we can still pass a `TimeId` to the constructor of `ObservationId`. 
- Keep the generic constructor of `ObservationId` and have it call a `.value()` member function if the `Id` template argument is not static-castable to `double`

**Comment by wthrowe** [comment]
Use `time.value()`.  This is a temporary solution until dense output is implemented.

**Comment by nilsvu** [comment]
@wthrowe I made the changes, this should be ready for review now

File: tests/Unit/IO/Observers/Test_ObservationId.cpp [hunk 1]
```diff
@@ -81,11 +55,11 @@ SPECTRE_TEST_CASE("Unit.IO.Observers.ObservationId", "[Unit][Observers]") {
   CHECK(id4.observation_type_hash() != id3.observation_type_hash());
   CHECK(id4.value() == 8.0);
 
-  ObservationId time0(Time(Slab{0.0, 1.0}, 0),
+  ObservationId time0(Time(Slab{0.0, 1.0}, 0).value(),
```
Comment (wthrowe) [inline tests/Unit/IO/Observers/Test_ObservationId.cpp]: This section (everything with the `timeN` objects) is a duplicate of the preceding section now.

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Comment by nilsvu** [comment]
fixed @wthrowe. @nilsdeppe and/or @markscheel please review

**Comment by wthrowe** [comment]
Looks good.

**Comment by nilsvu** [comment]
Rebased and squashed

**Comment by nilsvu** [comment]
Rebased on #1385

**Review by nilsdeppe — APPROVED** [review]

**Review by markscheel — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@wthrowe please look at; you previously gave the go ahead for a squash, so it should be quick; please merge if you are happy with it.

**Review by wthrowe — APPROVED** [review]

**Pull request merged**